src/pages/Flow/Actions.vue

```diff
@@ -117,45 +128,58 @@ export default {
           },
           errorPolicy: 'all'
         })
+
+        if (errors) throw new Error(errors?.[0]?.message)
+
+        this.isRunning = false
+
         if (data?.create_flow_run?.id) {
-          this.$router.push({
-            name: 'flow-run',
-            params: { id: data?.create_flow_run?.id, tenant: this.tenant?.slug }
+          const run = await this.$apollo.query({
+            query: require('@/graphql/Flow/flow-run-by-pk.gql'),
+            variables: {
+              id: data.create_flow_run.id
+            },
+            errorPolicy: 'all'
+          })
+
+          await this.updateNotification({
+            id: notificationId,
+            notification: {
+              color: 'primary',
+              loading: false,
+              linkText: 'Visit',
+              to: {
+                name: 'flow-run',
+                params: {
+                  id: data.create_flow_run.id,
+                  tenant: this.tenant?.slug
+                }
+              },
+              text: 'Run created!',
+              subtext: run?.data?.flow_run_by_pk?.name,
+              dismissable: true,
+              timeout: 20000
+            }
           })
-        }
-        if (errors) {
-          this.quickRunErrorAlert(errors[0]?.message)
         }
       } catch (err) {
-        this.quickRunErrorAlert()
+        await this.updateNotification({
+          id: notificationId,
+          notification: {
+            color: 'error',
+            loading: false,
+            text: 'There was a problem creating your run',
+            subtext: err?.message?.includes('archived')
+              ? 'This flow version is archived - you likely have a newer version of your flow'
+              : 'Error',
```

Comment:
Maybe this is one for wider discussion but I prefer adding the error message passed back by the API.  They're now written in a way that is meant to be user friendly and help us and users with de-bugging.  Either way, I'm not sure 'Error' is helpful - if we don't include the actual error message maybe just leave blank?

Reply:
Yeah I agree in most cases wrt API error messages but less so in application errors that we don't explicitly throw. In this particular instance though, the API is returning almost the same message but with an id embedded, which isn't necessary to show the user (they're already on the flow page!)